### PR TITLE
Migrate to jest/eslint

### PR DESCRIPTION
**Pull request by rmuchall**
Hi folks,

Following my other PRs to typestack here is a similar one for class-transformer.
https://github.com/typestack/routing-controllers/pull/553
https://github.com/typestack/class-validator/pull/573

**Action taken**
Replaced mocha, chai, sinon and nyc with jest
**Reasoning**
To normalize the test framework across all my projects
Jest appears to be a more complete framework
Jest is more popular
https://npmcompare.com/compare/jest,mocha

**Action taken**
Replaced tslint with eslint
**Reasoning**
tslint is deprecated and the recommended replacement is eslint

**Action taken**
Replaced gulp with npm scripts (copyfiles, json, rimraf)
**Reasoning**
Gulp is awesome but we can achieve the same result by using npm scripts (same as routing-controllers)

Thanks,
Rob

### CODE REVIEW

**Comment by jotamorais** [comment]
@NoNameProvided, I reviewed this PR in preparation to integrate other PRs and other important fixes (such as #339)

I cloned and integrated it locally but when I tried and to push to the develop branch, I noticed that the branch is protected and and the CI won't allow me to push. Would you please either add your review to the PR as well, or grant me permission to push to that branch (or remove that status check from Travis?  

Here's what I get:

image

Thanks!

**Comment by jotamorais** [comment]
> @NoNameProvided, I reviewed this PR in preparation to integrate other PRs and other important fixes (such as #339)
> 
> I cloned and integrated it locally but when I tried and to push to the develop branch, I noticed that the branch is protected and and the CI won't allow me to push. Would you please either add your review to the PR as well, or grant me permission to push to that branch (or remove that status check from Travis?
> 
> Here's what I get:
> 
> image
> 
> Thanks!

@NoNameProvided / @pleerock 
Appreciate if you could help me with this.

**Comment by saulotoledo** [comment]
@jotamorais, I was talking with @pleerock. Could you try again?

**Comment by jotamorais** [comment]
> @jotamorais, I was talking with @pleerock. Could you try again?

@saulotoledo & @pleerock,

I tried again but the branch is still protected and with the commit hook and won't allow me to push.

image

File: package.json [hunk 1]
```diff
@@ -26,40 +26,30 @@
...
+    "typescript": "3.8.3"
   },
   "scripts": {
-    "test": "gulp tests",
-    "package": "gulp package"
+    "build": "rimraf build && echo Using TypeScript && tsc --version && tsc --pretty",
+    "clean": "rimraf build coverage",
+    "copy": "copyfiles -u 3 \"build/compiled/src/**/*\" build/package && copyfiles package.json README.md build/package",
+    "lint": "eslint --config ./.eslintrc.js --ext .ts ./src ./test",
+    "package": "npm run build && npm run copy && npm run public && rimraf build/compiled",
```
Comment (saulotoledo) [inline package.json:50]: I suggest to use https://github.com/mysticatea/npm-run-all here to make it cleaner.

**Comment by saulotoledo** [comment]
@jotamorais Please find me on gitter if you have some time later.

**Comment by pleerock** [comment]
@jotamorais I don't watch for notifications in these repositories, since I don't maintain them. But you can reach me in email or typeorm's slack in case if you need something from me. @NoNameProvided I temporary switched number of approvals to master back to `1` sorry if I did something wrong, you can return it back to `2` once you have a time.

**Pull request merged**